codec-http2/src/main/java/io/netty/handler/codec/http2/DecoratingHttp2ConnectionEncoder.java

```diff
@@ -65,8 +65,9 @@ public void remoteSettings(Http2Settings settings) throws Http2Exception {
     public void consumeReceivedSettings(Http2Settings settings) {
         if (delegate instanceof Http2SettingsReceivedConsumer) {
             ((Http2SettingsReceivedConsumer) delegate).consumeReceivedSettings(settings);
+        } else {
+            throw new IllegalStateException("delegate " + delegate + " is not an instance of " +
```

Comment:
this should be class cast exception, right?

Reply:
Not sure 🤔